### ISSUE TITLE
Improve Sharedown link detection

### ISSUE THREAD

**Issue by minnas23** [kylon/Sharedown#99]
I use Sharedown with Yarn Run from source

### Run from source

```
yarn install
yarn start
```

When I add link to Sharedown, example
`https://xxx-my.sharepoint.com/personal/xxx/_layouts/15/stream.aspx?id=name&referrer=StreamWebApp%2EWeb&referrerScenario=AddressBarCopied%2Eview%2E0c7b2c8e%2D0805%2D4a0d%2D8fec%2D400d407d911f`

Chrome for Testing access link, and Sharepoint change `0c7b2c8e%2D0805%2D4a0d%2D8fec%2D400d407d911f` in link above to different value, and Sharedown can't detect link and see it as error. Have anyway to fix it?

Such as change Sharedown to support shortened links and when user press "Start Download", Chrome for Testing will access the link and will download the video detected there instead of just downloading the video in an absolute link value.

**Comment by minnas23** [comment]
I think you should add "Error" beside "Download Queue" and "Complete" and add "Ignore Error" to Setting.

Example:
`(10) Download queue - (0) Complete - (2) Error`

Because even though the download failed but with "Do not delete temp directory on fail", I saw the video downloaded at "Fragments temp directory". With "Ignore Error", Sharedown can skip the error videos and download the remaining videos in Queue. Video Error will go to "Error" with "Download queue" and "Complete"

**Comment by kylon** [comment]
are you getting the link from sharepoint context menu?

yea, i ll do something for that

**Comment by minnas23** [comment]
No, this is direct link of video. When video played, Sharepoint changed value of link to other value. Sharedown can't detect this change, but yt-dlp still download, and when video downloaded, sharedown see this as error, but video still downloaded in "Fragments temp directory". And if I turn on "retry download video if download error", sharedown not download other video in queue and redownload error video with infinity loop.

**Comment by kylon** [comment]
retry option is a loop, its not a bug
copy the link from sharepoint context menu, this is the expected link format

**Comment by minnas23** [comment]
https://github.com/kylon/Sharedown/issues/92#issuecomment-2067964916

can you give this file to me? I can't open it. 
When I give this link of folder to "Import URLs from folder", Sharedown can't detect this link and return error in image.
![image](https://github.com/user-attachments/assets/27bc2710-67d2-4ba4-a994-0afb1864c1ec)

Here is format of folder link
https://xxx-my.sharepoint.com/personal/user_xxx_onmicrosoft_com/_layouts/15/onedrive.aspx?ga=1&id=folder_id

**Comment by kylon** [comment]
i will upload a test later today

**Comment by kylon** [comment]
a bit late, a lightning fried my isp

i dont know if that is a pure sharepoint folder or if something has changed recently (in which case, since i no longer use ms apps, i would wait for someone to add support for new stuff)

the only url i can get in sharepoint is
`https://xx-my.sharepoint.com/:f:/g/personal/xx/Eoy732JHMr5KhFt4fHmUwmkB6-XWTM-FYLP_BLBzLZWttw?e=FzkAng`


anyway, try to import folder with this
[sharedownt.zip](https://github.com/user-attachments/files/16912788/sharedownt.zip)

run from terminal, `npm -i` in extracted folder to install packages, `npm start` to run


i tried a link like the one in first post and it worked, can you post a log?

**Comment by minnas23** [comment]
Can i contact you via email or other messaging channels? I want give you my sharedown folder i use, a video how i access my sharepoint folder, and my direct sharepoint link via email. i think only log file can't solve this issue.

**Comment by kylon** [comment]
mh, if it requires login it may be a problem, did you try the test above?
discord?
